contracts/Controller.sol

```diff
@@ -9,6 +9,13 @@ import "zeppelin-solidity/contracts/lifecycle/Pausable.sol";
 contract Controller is Pausable, IController {
     // Track contract ids and their mapped addresses
     mapping (bytes32 => address) registry;
+    // Track function permissions on registered contracts
+    mapping (address => mapping (address => mapping (bytes4 => bool))) permissions;
```

Comment:
the first two keys (address[address])....are they `owner_address[contract_address]`, or `contract_address[owner_address]`?